File: app/src/main/java/tr/name/fatihdogan/books/dependecy/module/AppModule.java
package tr.name.fatihdogan.books.dependecy.module;

import android.app.Application;
import android.content.Context;
import android.view.inputmethod.InputMethodManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private final Application application;

    public AppModule(Application application) {
        this.application = application;
    }

    @Provides
    @Singleton
    Application providesApplication() {
        return application;
    }

    @Provides
    @Singleton
    InputMethodManager inputMethodManager(Application application) {
        return (InputMethodManager) application.getSystemService(Context.INPUT_METHOD_SERVICE);
    }
}

File: app/src/main/java/com/fthdgn/books/dependecy/module/AppModule.kt
package com.fthdgn.books.dependecy.module

import android.app.Application
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(private val application: Application) {

    @Provides
    @Singleton
    internal fun providesApplication(): Application =
            application
}
